app/helpers/automated_tests_helper.rb

```diff
@@ -119,114 +119,52 @@ def create_ant_test_files(assignment)
   # - Process new and updated test files (additional validation to be done at the model level)
   def process_test_form(assignment, params)
```

Comment:
Did you leave a comment earlier? Add in an extra parameter here: `process_test_form(assignment, params, new_files)` where the last one is obtained directly from `params`.